src/components/layerselector/LayerSelectorDirective.js

```diff
@@ -0,0 +1,147 @@
+(function() {
+  goog.provide('ga_layerselector_directive');
+
+  goog.require('ga_map');
+  goog.require('ga_permalink');
+
+  var module = angular.module('ga_layerselector_directive', [
+    'ga_map',
+    'ga_permalink',
+    'pascalprecht.translate'
+  ]);
+
+  module.directive('gaLayerSelector',
+    function(gaPermalink, gaLayers, gaLayerFilters) {
+      return {
+        restrict: 'A',
+        templateUrl: 'components/layerselector/partials/layerselector.html',
+        scope: {
+          map: '=gaLayerSelectorMap'
+        },
+        link: function(scope, elt, attrs) {
+          scope.isClosed = true;
+          var map = scope.map;
+          var firstLayerChangeEvent = true;
+          // Background layers are always the same
+          scope.backgroundLayers = [
```

Comment:
`var backgroundLayers = gaLayers.getBackgroundLayers().getArray();` and then `backgroundLayers.push({id: 'voidLayer', label: 'void_layer'});` as it was done previously.
If we had/remove a bg layer we don't need to change the code.